metadata-ingestion/src/datahub/ingestion/graph/client.py

```diff
@@ -957,16 +959,62 @@ def delete_references_to_urn(
 
     @functools.lru_cache()
     def _make_schema_resolver(
-        self, platform: str, platform_instance: Optional[str], env: str
+        self,
+        platform: str,
+        platform_instance: Optional[str],
+        env: str,
+        include_graph: bool = True,
     ) -> "SchemaResolver":
         from datahub.utilities.sqlglot_lineage import SchemaResolver
 
         return SchemaResolver(
             platform=platform,
             platform_instance=platform_instance,
             env=env,
-            graph=self,
+            graph=self if include_graph else None,
+        )
+
+    def initialize_schema_resolver_from_datahub(
+        self, platform: str, platform_instance: Optional[str], env: str
+    ) -> Tuple["SchemaResolver", Set[str]]:
+        logger.info("Initializing schema resolver")
+
+        # TODO: Filter on platform instance?
+        logger.info(f"Fetching urns for platform {platform}, env {env}")
+        with PerfTimer() as timer:
+            urns = set(
+                self.get_urns_by_filter(
+                    entity_types=[DatasetUrn.ENTITY_TYPE],
+                    platform=platform,
```

Comment:
there's a bug here - it doesn't respect platform_instance

Reply:
So it won't actually break anything right -- just add more items to schema resolver than necessary. Do we have the ability to filter on platform instance?

Reply:
Yep - I'm adding that here https://github.com/datahub-project/datahub/pull/8709